public/app/features/plugins/sandbox/sandbox_plugin_loader.ts

```diff
@@ -74,6 +74,10 @@ async function doImportPluginModuleInSandbox(meta: PluginMeta): Promise<unknown>
       liveTargetCallback: isLiveTarget,
       // endowments are custom variables we make available to plugins in their window object
       endowments: Object.getOwnPropertyDescriptors({
+        // window.location is unforgeable, we make the location available via endowments
+        // when the plugin code is loaded, the sandbox replaces the window.location with
+        // window.locationSandbox. In the future this can be a proxy
```

Comment:
I'll expand the comment to clarify. What I meant here is that instead of directly exposing window.location inside window.locationSandbox we can expose a proxyfied version of window.location so we can even control what kind of `location` the plugins have access.